components/director/pkg/operation/update_operation_handler.go

```diff
@@ -182,3 +186,33 @@ func stringifiedJsonError(errorMsg string) (*string, error) {
 	stringifiedErr := string(bytesErr)
 	return &stringifiedErr, nil
 }
+
+func determineApplicationFinalStatus(opType OperationType, opError *string) model.ApplicationStatusCondition {
+	appConditionStatus := model.ApplicationStatusConditionInitial
+	switch opType {
+	case OperationTypeCreate:
+		appConditionStatus = model.ApplicationStatusConditionCreateSucceeded
+		if opError != nil || ptrStrToStr(opError) != "" {
+			appConditionStatus = model.ApplicationStatusConditionCreateFailed
+		}
+	case OperationTypeUpdate:
+		appConditionStatus = model.ApplicationStatusConditionUpdateSucceeded
+		if opError != nil || ptrStrToStr(opError) != "" {
+			appConditionStatus = model.ApplicationStatusConditionUpdateFailed
+		}
+	case OperationTypeDelete:
+		appConditionStatus = model.ApplicationStatusConditionDeleteSucceeded
+		if opError != nil || ptrStrToStr(opError) != "" {
+			appConditionStatus = model.ApplicationStatusConditionDeleteFailed
+		}
+	}
+
+	return appConditionStatus
+}
+
+func ptrStrToStr(s *string) string {
```

Comment:
Move this function variation to this file: https://github.com/kyma-incubator/compass/blob/6e6973361cc9940351bbe3f7ba99ec9869e778e3/components/director/pkg/str/strings.go#L33